utils/log/logger.go

```diff
@@ -161,24 +194,50 @@ type Log interface {
 }
 
 // Check if Stderr is a terminal
-func isTerminalMode() bool {
+func IsTerminal() bool {
 	if terminalMode == nil {
 		t := term.IsTerminal(int(os.Stderr.Fd()))
 		terminalMode = &t
 	}
 	return *terminalMode
 }
 
+// IsColorsSupported returns true if the process environment indicates color output is supported and desired.
+func IsColorsSupported() bool {
+
+	if colorsSupported == nil {
+		supported := true
+
+		if !IsTerminal() ||
+
+			// https://en.wikipedia.org/wiki/Computer_terminal#Dumb_terminals
+			os.Getenv("TERM") == "dumb" ||
+
+			// On Windows WT_SESSION is set by the modern terminal component.
+			// Older terminals have poor support for UTF-8, VT escape codes, etc.
+			runtime.GOOS == "windows" && os.Getenv("WT_SESSION") == "" ||
```

Comment:
Mayne better to put parentheses around the Windows part in the if statement. Change - 
```runtime.GOOS == "windows" && os.Getenv("WT_SESSION") == "" ||```
into - 
```(runtime.GOOS == "windows" && os.Getenv("WT_SESSION")) == "" ||```